Mathlib/RingTheory/HahnSeries/Basic.lean

```diff
@@ -108,8 +108,19 @@ nonrec theorem support_nonempty_iff {x : HahnSeries Γ R} : x.support.Nonempty 
 theorem support_eq_empty_iff {x : HahnSeries Γ R} : x.support = ∅ ↔ x = 0 :=
   Function.support_eq_empty_iff.trans coeff_fun_eq_zero_iff
 
+/-- The map of Hahn series induced by applying a zero-preserving map to each coefficient. -/
+@[simps]
+def map [Zero S] (x : HahnSeries Γ R) (f : ZeroHom R S) : HahnSeries Γ S where
```

Comment:
Should this allow for `f` to come from an arbitrary `ZeroHomClass`? I see that it works fine for monoid and ring homs elsewhere, presumably using a coercion, so I'm not sure which implementation is better.

It's probably worth mentioning in the module doc that this maps coefficients, where `HahnSeries.embDomain` maps indices.

Reply:
Thank you, that had not occurred to me.